tests/test_measurements.py

```diff
@@ -391,6 +390,22 @@ def circuit():
         with pytest.raises(qml.QuantumFunctionError, match="CNOT is not an observable"):
             sample = circuit()
 
+    def test_providing_symbolic_operation_raises_error(self):
+        """Test that a qml.QuantumFunctionError is raised if the provided
+        argument is a symbolic operation"""
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX(0.52, wires=0)
+            return qml.sample(qml.op_sum(qml.PauliX(0), qml.PauliZ(0)))
```

Comment:
Was this test pulled in from another branch? This seems to be testing `qml.sample`.

Reply:
This test was added for both `qml.sample` and `qml.count` to cover the line (which is now present in both functions):
https://github.com/PennyLaneAI/pennylane/blob/1558f7d561b0cc709630e2f9d030101afa873b36/pennylane/measurements.py#L686

Reply:
I just realised that this line is covered in `test_sum.py` and `test_sprod.py`:
https://github.com/PennyLaneAI/pennylane/blob/1558f7d561b0cc709630e2f9d030101afa873b36/tests/ops/op_math/test_sum.py#L695
I will add another test to cover the error for the `qml.count` method.